Added connect overview page. (#394)
wormhole-foundation/wormhole-docs: products/connect/overview.md

For developers seeking UI customization, the [Wormhole TypeScript SDK](docs/build/toolkit/typescript-sdk/){target=\_blank} provides a powerful alternative to the Connect widget, empowering you to integrate the same core functionality directly into your own user interface. Refer to the SDK documentation for comprehensive usage instructions.

## Features

Wormhole Connect offers flexible customization to match your application. Tailor technical aspects like supported assets and custom RPCs, or use the full-featured default widget. Its UI is also highly adaptable, offering extensive styling options, including a no-code visual interface. Connect’s features include:

- Multiple ways to bridge assets (routes)
- Extensive ways to style the UI (including the no-code styling interface)
- Ways to configure what feature set to offer
- Ability to configure any token to bridge via Wormhole
- Ability to drop off some gas at the destination

Be sure to check the [features page](docs/build/transfers/connect/features/){target=\_blank} for more details about Wormhole Connect's features and a breakdown of supported features by chain.

## How It Works

> Think of Wormhole Connect as a ready-made interface that simplifies the process of bridging assets across different blockchains. When a user initiates a transfer using the widget:

Reviewer: Hmmmm... this section is a bit interesting for a tool like Connect. Wdyt about this @ilariae @martin0995 @dawnkelly09 ?

Reviewer 2: I see your point, but I think it's also nice to have it for someone new who knows nothing about Connect. Let's link somewhere to the Portal Bridge, so the reader can directly go and play with it.

Reviewer 3: I'm conflicted about this "when a user initiates a transfer using the widget" bit for two reasons:

1. All of these steps may or may not happen every time, depending on how things are configured
2. There isn't anything about this described process that is unique to Connect. It's just how transfers work

I think Connect's "How It Works" section should focus on what it makes easier for the developer. They get a customizable UI that allows users to interact with their WH integration without all the overhead required to build that UI manually, load all the needed data to configure each supported chain, etc.

/>
```
For example, for 0.3.13:

```HTML
<!-- Mounting point. Include in <body> -->
<div id="wormhole-connect"></div>

<!-- Dependencies -->
<script
    type="module"
    src="https://www.unpkg.com/@wormhole-foundation/wormhole-connect@0.3.13/dist/main.js"
    defer
></script>
<link
    rel="https://www.unpkg.com/@wormhole-foundation/wormhole-connect@0.3.13/dist/main.css"
/>
```
It is crucial to periodically update your Wormhole Connect instance to the latest version, as frequent functionality and security updates are released.

## Configuration

This section provides a brief overview of the configuration possibilities. For comprehensive details on all available configuration options, please consult the Configuration docs.

The default configuration of Wormhole Connect might not perfectly align with your specific requirements. You might want to:

- Implement custom styles
- Restrict the chains that are supported within your application
- Add support for your project's token and remove irrelevant tokens to minimize clutter
- Configure custom RPC URLs (This is highly recommended as default public RPCs often experience heavy throttling)
- Limit the available bridging routes

> For further information on these options, please refer to the [configuration options](docs/build/transfers/connect/configuration/){target=\_blank} to customize your widget according to your preferences.

Reviewer: This sort of stuff would be in the get started page instead of the overview page

> ## Next Steps 

Reviewer: Definitely should push them to the get started guide

## How It Works

Think of Wormhole Connect as a ready-made interface that simplifies the process of bridging assets across different blockchains. When a user initiates a transfer using the widget:

1.  **Initiation** - select the asset, the originating blockchain, and the target blockchain for the transfer
2.  **Connection** - establish a link between your wallet and the chosen source blockchain
3.  **Transaction submission on source chain** - confirm the transfer details to trigger the asset lock or deposit on the initial blockchain
4.  **Wormhole message creation** - Wormhole's network observes the source transaction, and Guardians validate it to generate the cross-chain proof 
5.  **Message relaying** -  the generated CCT or VAA is transmitted across the Wormhole network to the intended destination blockchain
6.  **Transaction on destination chain** - Wormhole's contracts on the target blockchain receive and verify the incoming CCT or VAA
7.  **Asset release/minting** - upon successful verification, the equivalent assets are either released as wrapped assets  or newly created on the destination blockchain and delivered to your wallet

## Use Cases

> Here are key use cases that highlight the power and versatility of the Token Bridge.

Reviewer 3: ```suggestion
Here are key use cases that highlight the power and versatility of Connect.
```

Reviewer: This change was marked as resolved here: https://github.com/wormhole-foundation/wormhole-docs/pull/394#discussion_r2096611736. So please make sure you're not resolving conversations until after the fix has been applied so we don't lose track of these items! 🙂

---
title: Wormhole Connect
description: With Wormhole Connect, you can seamlessly bridge digital assets and data across a wide range of supported blockchain networks.
categories: Connect, Transfer

---

# Wormhole Connect Overview 

> With the Wormhole Connect React widget, you can enable your users to perform cross-chain asset transfers directly within your web application, powered by the Wormhole protocol's easy-to-use interface.

Reviewer: It can also be embedded into any javascript project, not just React, using the hosted version: https://wormhole.com/docs/build/transfers/connect/overview/#use-hosted-version-via-cdn

Reviewer: One sentence seems just a bit too short. So, what is the core problem it solves? Or the value it delivers? I think we can add a sentence or two that answers these questions.

> With the Wormhole Connect widget, you can enable your users to perform cross-chain asset transfers directly within your web application, powered by the Wormhole protocol's easy-to-use interface. Connect simplifies the complexity of bridging, offering a single, intuitive point of interaction for moving assets across diverse blockchains. This empowers you to seamlessly access liquidity and opportunities across any connected network.

Reviewer 4: ```suggestion
With the Wormhole Connect widget, you can enable users to perform multichain asset transfers directly within your application. Connect simplifies the complexity of bridging, offering a single, intuitive point of interaction for moving assets across diverse blockchains. This empowers you to access liquidity and opportunities across any connected network seamlessly.

```

Wormhole Connect offers flexible customization to match your application. Tailor technical aspects like supported assets and custom RPCs, or use the full-featured default widget. Its UI is also highly adaptable, offering extensive styling options, including a no-code visual interface. Connect’s features include:

- **In-app multichain transfers** – bridge assets without leaving your app
- **Flexible bridging routes** – supports multiple transfer pathways
- **Customizable UI** – style the bridge interface to match your brand 
- **Customizable features** – select bridging options 
- **Wormhole token bridging** – secure cross-chain transfers via Wormhole
- **Optional destination gas** – provide gas for initial transactions on the target chain

Be sure to check the [features page](/docs/products/connect/reference/support-matrix/#feature-support-matrix){target=\_blank} for more details about Wormhole Connect's features and a breakdown of supported features by chain.

## How It Works

> Think of Connect as a ready-made interface that simplifies the process of bridging assets across different blockchains. As you begin a cross-chain asset move, Connect simplifies the steps by:

Reviewer 4: ```suggestion
Think of Connect as a ready-made interface that simplifies the process of bridging assets across different blockchains. As you begin a multichain transfer, Connect simplifies the steps by:
```

> 1.  **Initiation** - selects the asset, the originating blockchain, and the target blockchain for the transfer

Reviewer 4: ```suggestion
1.  **Initiation** - selects the asset, the source chain, and the target chain for the transfer
```

you can use chain without repeating blockchain every time
usually we call them source chains and target chains

> 2.  **Connection** - establishes a link between your wallet and the chosen source blockchain

Reviewer 4: ```suggestion
2.  **Connection** - connect your chosen wallet to the source chain
```

> 3.  **Transaction submission on source chain** - confirms the transfer details to trigger the asset lock or deposit on the initial blockchain

Reviewer 4: I think there's something missing here from connecting wallet to submitting 
you can refer to this tutorial for the flow https://wormhole.com/docs/tutorials/connect/react-dapp/

> 4.  **Wormhole message creation** - Wormhole's network observes the source transaction, and Guardians validate it to generate the cross-chain proof 

Reviewer 4: ```suggestion
4.  **Wormhole message creation** - Wormhole's network observes the source transaction, and Guardians validate it to generate the VAA
```

use VAA when referring to the guardian's proof - u can also add a link to VAAs here

4.  **Wormhole Message Creation** - Wormhole's network observes the source transaction, and Guardians validate it to generate the cross-chain proof 
5.  **Message Relaying** -  the generated CCT or VAA is transmitted across the Wormhole network to the intended destination blockchain
6.  **Transaction on Destination Chain** - Wormhole's contracts on the target blockchain receive and verify the incoming CCT or VAA
7.  **Asset Release/Minting** - upon successful verification, the equivalent assets are either released as wrapped assets  or newly created on the destination blockchain and delivered to your wallet

> ## Use Cases

Reviewer: All the links in this section need to be updated

> Here are key use cases that highlight the power and versatility of the Token Bridge.

Reviewer: Haha I'm assuming this was forgotten about. Just a friendly reminder to not forget to update it 🙂

> 5.  **Message relaying** -  autmomates relays of the generated CCT or VAA across the Wormhole network to the intended destination blockchain

Reviewer 4: CCT?

> 6.  **Transaction on destination chain** - contracts on the target blockchain receive and verify the incoming CCT or VAA

Reviewer 4: What's CCT
if you mean CCTP i don't think we should treat it as a VAA bc cctp is the protocol not a proof itself
this is the connect page so just use VAA and forget about cctp here

- **Cross-Chain Swaps and Liquidity Aggregation**

    - [**Wormhole Connect**](/docs/build/transfers/connect/overview/) – handles user-friendly asset transfers

>     - [**Native Token Transfer**](/docs/build/transfers/native-token-transfers/) – moves native assets across chains

Reviewer: ```suggestion
    - [**Native Token Transfers**](/docs/build/transfers/native-token-transfers/) – moves native assets across chains
```

> 7.  **Asset release/minting** - upon successful verification, the equivalent assets are either released as wrapped assets  or newly created on the destination blockchain and delivered to your wallet

Reviewer 4: ```suggestion
7.  **Asset release/minting** - upon successful verification, the equivalent assets are either released as wrapped assets or newly created on the destination blockchain and delivered to your wallet
```

> _Note_ If you want more hands on experience with Connect, checkout Wormhole [Portal Bridge](https://portalbridge.com/).

Reviewer 4: ```suggestion
!!!note
If you want more hands-on experience with Connect, check out Wormhole [Portal Bridge](https://portalbridge.com/)
```

check that this renders, there should be a tab space before the text but it won't let me add it

    - [**Queries**](/docs/build/queries/overview/) – fetches real-time prices for optimal trade execution

- **Cross-Chain Payment Widgets**

    - [**Wormhole Connect**](/docs/build/transfers/connect/overview/) – facilitates seamless payments in various tokens

>     - [**Native Token Transfer**](/docs/build/transfers/native-token-transfers/) – ensures direct, native asset transfers

Reviewer: ```suggestion
    - [**Native Token Transfers**](/docs/build/transfers/native-token-transfers/) – ensures direct, native asset transfers
```

>     - [**Wormhole Connect**](/docs/products/connect/get-started/) – handles user-friendly asset transfers
>     - [**Native Token Transfers**](/docs/products/native-token-transfers/overview/) – moves native assets across chains
>     - [**Queries**](/docs/products/queries/overview/) – fetches real-time prices for optimal trade execution
> 
> - **Cross-Chain Payment Widgets**
> 
>     - [**Wormhole Connect**](/docs/products/connect/get-started/) – facilitates seamless payments in various tokens
>     - [**Native Token Transfers**](/docs/products/native-token-transfers/overview/) – ensures direct, native asset transfers
> 
> - **Web3 Game Asset Transfers**
> 
>     - [**Wormhole Connect**](/docs/products/connect/get-started/) – provide a user-friendly way to move game tokens across chains
>     - [**Token Bridge**](/docs/products/token-bridge/overview/) – handle the underlying lock-and-mint logic securely

Reviewer: ```suggestion
    - [**Connect**](/docs/products/connect/get-started/) – handles user-friendly asset transfers
    - [**Native Token Transfers**](/docs/products/native-token-transfers/overview/) – moves native assets across chains
    - [**Queries**](/docs/products/queries/overview/) – fetches real-time prices for optimal trade execution

- **Cross-Chain Payment Widgets**

    - [**Connect**](/docs/products/connect/get-started/) – facilitates seamless payments in various tokens
    - [**Native Token Transfers**](/docs/products/native-token-transfers/overview/) – ensures direct, native asset transfers

- **Web3 Game Asset Transfers**

    - [**Connect**](/docs/products/connect/get-started/) – provide a user-friendly way to move game tokens across chains
    - [**Token Bridge**](/docs/products/token-bridge/overview/) – handle the underlying lock-and-mint logic securely
```

## Next Steps 

> Do you feel ready to get started with Connect? Explore these pages to get started:

Reviewer: ```suggestion
Add Connect to your app with these key setup steps:

```
